client/director.go

```diff
@@ -164,17 +93,7 @@ func queryDirector(ctx context.Context, verb, sourcePath, directorUrl string) (r
 		}
 	}
 
-	// If we get a 404, the director will hopefully tell us why. It might be that the namespace doesn't exist
-	if resp.StatusCode == 404 && verb == "PROPFIND" {
```

Comment:
Quite a few of these errors deal with falling back to a "dirlisthots" for the given namespace -- a concept that comes from Topology and isn't something we'd get from any modern Director. And while the Director previously supplied a `dirlisthost` in its `X-Pelican-Namespace` header, this hasn't been the case for many, many months. After 4 or 5 major releases, I think we're safe dropping the backwards compatibility for this edge case, especially because we keep our central services up to date and I don't think there's anyone out there running something this old.